### DISCUSSION TITLE
How do I get the value of child after the date element

### DISCUSSION THREAD

**Discussion by chinmaypatil95** [nlohmann/json#2617] [Q&A] [answered]
Hello,
          I have following json schema. I need to extract the value of indicator which is after a date element (only for the latest date - single value). Can anyone help me to get the value of latest indicator into a string ?
          
Thanks.

```json
{
    "Meta Data": {
        "ABC": "123",
        "XYZ": "456",
        "PQR": "789"
    },
    "Sensor1": {
        "2021-01-29": {
            "Indicator": "33.3002"
        },
        "2021-01-28": {
            "Indicator": "40.8689"
        },
        "2021-01-27": {
            "Indicator": "47.6354"
        },
        "2021-01-25": {
            "Indicator": "54.2591"
        }
    }
}
```

**Comment by nlohmann** [comment, marked as answer]
JSON objects do not have an order, so conceptually there is no "first" or "last" element. This library, however, sorts keys alphabetically (it uses a `std::map<std::string, json>` internally), so assuming the "latest date" is the alphabetically greatest element, then

```cpp
*j["Sensor"].rbegin();
```

should be  reference to that value, so

```json
{
    "Indicator": "33.3002"
}
```

in your case.

**Reply by nlohmann** [reply]
@chinmaypatil95 Do you need further assistance with this issue?
